configure.ac

```diff
@@ -241,38 +241,44 @@ AC_SEARCH_LIBS([clock_gettime], [rt],
 dnl check for libraries
 
 OMPI_CHECK_PORTALS4([portals4],
-    [AC_DEFINE([USE_PORTALS4], [1], [Define if Portals 4 transport active])
-     transport="portals4"
-     transport_portals4="yes"],
+    [transport_portals4="yes"],
     [transport_portals4="no"])
-AM_CONDITIONAL([USE_PORTALS4], [test "$transport_portals4" = "yes"])
 
-OPAL_CHECK_LIBFABRIC([libfabric],
-    [AC_DEFINE([USE_OFI], [1], [Define if OFI transport active])
-     transport="ofi"
-     transport_ofi="yes"],
+OPAL_CHECK_OFI([ofi],
+    [transport_ofi="yes"],
     [transport_ofi="no"])
-AM_CONDITIONAL([USE_OFI], [test "$transport_ofi" = "yes"])
 
-if test "$transport_ofi" = "no" ; then
-  OMPI_CHECK_OFI([ofi],
-      [AC_DEFINE([USE_OFI], [1], [Define if OFI transport active])
-       transport="ofi"
-       transport_ofi="yes"],
-      [transport_ofi="no"])
-  AM_CONDITIONAL([USE_OFI], [test "$transport_ofi" = "yes"])
+# assure that only ofi is enabled if portals4 is also detected
+if test -n "$with_ofi" -a "$with_ofi" != "no" ; then
+    transport="ofi"
+    transport_portals4="no"
+    AC_DEFINE([USE_OFI], [1], [Define if OFI transport active])
+elif test -n "$with_portals4" -a "$with_portals4" != "no" ; then
+    transport="portals4"
+    transport_ofi="no"
+    AC_DEFINE([USE_PORTALS4], [1], [Define if Portals 4 transport active])
+else
+    AC_MSG_WARN([Neither OFI nor portals4 transport found])
```

Comment:
I think it would be more accurate for the message to say "requested" rather than "found".  It's possible to reach this stage with both OFI and Portals having been detected, but neither requested by the user.  Should we add an error case here for: ```test "$transport_ofi" = "yes" -a "$transport_portals4" = "yes"```?